### DISCUSSION TITLE
Return custom exit code

### DISCUSSION THREAD

**Discussion by mwadders** [PSAppDeployToolkit/PSAppDeployToolkit#701] [General]
I've been trying to get the script to return an exit code of 3010 after uninstall to force sccm to restart.  I've hard-coded it after the uninstall has run through and also in the exit-script function but it never seems to get returned. 
_Exit-Script -ExitCode 3010_
 
I'm obviously doing something wrong here so any pointers would be greatly appreciated.

**Comment by paulwetter** [comment]
What does the log say it is completing with?  Have you tried appending “-allowrebootpassthru $true” to your command line in CM app?

**Reply by mwadders** [reply]
Cheers for the advice, it looks like -allowrebootpassthru is the way to go.

**Comment by That-Annoying-Guy** [comment]
PSADT sometimes changes the Exitcode b/c SCCM is dumb.

FYI: you might have more eyeballs looking at this by posting here: https://discourse.psappdeploytoolkit.com/c/thetoolkit/general-discussion

**Comment by mwadders** [comment]
Nice one TAG, https://discourse.psappdeploytoolkit.com/t/exit-script-exitcode-3010-does-not-work/2320 seems to provide the answer I'm looking for.
